src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -57,14 +58,25 @@ public ActionResult execute() throws EntityDoesNotExistException {
             
             boolean isSectionChanged = student.isSectionChanged(originalStudentAttribute);
             boolean isTeamChanged = student.isTeamChanged(originalStudentAttribute);
+            boolean isEmailChanged = student.isEmailChanged(originalStudentAttribute);
             if (isSectionChanged) {
                 logic.validateSectionsAndTeams(Arrays.asList(student), courseId);
             } else if (isTeamChanged) {
                 logic.validateTeams(Arrays.asList(student), courseId);
             }
             
             logic.updateStudent(studentEmail, student);
-            statusToUser.add(new StatusMessage(Const.StatusMessages.STUDENT_EDITED, StatusMessageColor.SUCCESS));
+            
+            boolean isSendEmail = Boolean.parseBoolean(getRequestParamValue(Const.ParamsNames.EMAIL_SEND_CHECK));
+            if (isEmailChanged && isSendEmail) {
+                FeedbackSessionsLogic.inst().sendAllLinksToNewStudentEmail(courseId, student);
```

Comment:
Do not call `FeedbackSessionsLogic` directly here. Action classes should communicate via the API class `Logic`, i.e `logic.sendAllLinks...`.